src/distributor-detector.ts

```
@@ -39,4 +40,73 @@ export class DistributorDetector {
       return false;
     }
   }
+
+  /**
+   * Parses an OwnerActs event log to extract distributor creation information.
+   *
+   * @param log - The ethers.Log object containing the event data
+   * @param blockTimestamp - The timestamp of the block containing the event
+   * @returns DistributorInfo object if the event is a valid distributor creation, null otherwise
+   */
+  static parseDistributorCreation(
+    log: ethers.Log,
+    blockTimestamp: number,
+  ): DistributorInfo | null {
+    try {
+      // Parse the OwnerActs event
+      const iface = new ethers.Interface(OWNER_ACTS_EVENT_ABI);
+      const parsedLog = iface.parseLog(log);
+
+      if (!parsedLog) {
+        return null;
+      }
+
+      // Extract event arguments
+      const method = parsedLog.args["method"];
+      const owner = parsedLog.args["owner"];
+      const eventData = parsedLog.args["data"];
+
+      // Check if this is a distributor creation method
+      const distributorType = this.getDistributorType(method);
+      if (!distributorType) {
+        return null;
+      }
+
+      // Validate data field has minimum length for method selector (4 bytes = 8 hex chars + 0x)
+      const METHOD_SELECTOR_LENGTH = 10; // "0x" + 8 hex chars
+      if (eventData.length < METHOD_SELECTOR_LENGTH) {
+        return null;
+      }
+
+      // Validate that the data contains a valid address after the method selector
+      try {
+        ethers.AbiCoder.defaultAbiCoder().decode(
+          ["address"],
+          "0x" + eventData.substring(METHOD_SELECTOR_LENGTH),
+        );
+      } catch {
+        // Data doesn't contain a valid address
+        return null;
```

It must always contain a valid address for the events we care about - so this should throw an error